e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,323 @@
+		"--keyring-backend", keyring.BackendTest,
+		"--output", "json",
+		"--yes",
+	}
+
+	// ACTION: Execute an authz SEND using a TF token from a grantee who is blacklisted
+	// EXPECTED: Request fails; Even though grantee is acting on behalf of the granter,
+	// the granter still cannot execute `send` due to being blacklisted
+	// Status:
+	// 	Granter1 has authorized Grantee1 to send 100usdc from their wallet
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, grantee)
+
+	_, err = val.AuthzExec(ctx, grantee, nestedCmd)
+	require.ErrorContains(t, err, fmt.Sprintf("an address (%s) is blacklisted and can not receive tokens: unauthorized", grantee.FormattedAddress()))
+
+	bal, err := noble.GetBalance(ctx, receiver.FormattedAddress(), "uusdc")
+	require.NoError(t, err)
+	require.True(t, bal.IsZero())
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, grantee)
+
+	// ACTION: Execute an authz SEND using a TF token from a granter who is blacklisted
+	// EXPECTED: Request fails; Granter is blacklisted
+	// Status:
+	// 	Granter1 has authorized Grantee1 to send 100usdc from their wallet
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, granter)
+
+	preSendBal := bal
+
+	_, err = val.AuthzExec(ctx, grantee, nestedCmd)
+	require.ErrorContains(t, err, fmt.Sprintf("an address (%s) is blacklisted and can not send tokens: unauthorized", granter.FormattedAddress()))
+
+	bal, err = noble.GetBalance(ctx, receiver.FormattedAddress(), "uusdc")
+	require.NoError(t, err)
+	require.True(t, bal.IsZero())
+	// bal should not change
+	// require.Equal(t, preSendBal, bal)
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, granter)
+
+	// ACTION: Execute an authz SEND using a TF token to a receiver who is blacklisted
+	// EXPECTED: Request fails; Granter is blacklisted
+	// Status:
+	// 	Granter1 has authorized Grantee1 to send 100usdc from their wallet
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, receiver)
+
+	preSendBal = bal
+
+	_, err = val.AuthzExec(ctx, grantee, nestedCmd)
+	require.ErrorContains(t, err, fmt.Sprintf("an address (%s) is blacklisted and can not receive tokens: unauthorized", receiver.FormattedAddress()))
+
+	bal, err = noble.GetBalance(ctx, receiver.FormattedAddress(), "uusdc")
+	require.NoError(t, err)
+	// bal should not change
+	require.Equal(t, preSendBal, bal)
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, receiver)
+
+	// ACTION: Execute an authz SEND using a TF token while the TF is paused
+	// EXPECTED: Request fails; chain is paused
+	// Status:
+	// 	Granter1 has authorized Grantee1 to send 100usdc from their wallet
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	preSendBal = bal
+
+	_, err = val.AuthzExec(ctx, grantee, nestedCmd)
+	require.ErrorContains(t, err, "the chain is paused")
+
+	bal, err = noble.GetBalance(ctx, receiver.FormattedAddress(), "uusdc")
+	require.NoError(t, err)
+	// bal should not change
+	require.Equal(t, preSendBal, bal)
+}
```

Comment:
Is there any value in adding a happy path test here (no accounts blacklisted, chain not paused)?

Reply:
I think so, mainly to validate the test. Added:
[`b345e59` (#373)](https://github.com/noble-assets/noble/pull/373/commits/b345e59ae591f5c7add6576d727a19e7c51c2c1a)